common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java

```diff
@@ -33,65 +63,120 @@ protected void initializeSharedPreferencesFileManager(String fileName) {
+        }
+
+        ListIterator<IShareSingleSignOnState> otherCaches = mSharedSSOCaches.listIterator();
+
+        while(otherCaches.hasNext()){
+            otherCaches.next().setSingleSignOnState(account, refreshToken);
         }
 
         // TODO: I'd like to know exactly why this is here before I put this back in.... i'm assuming for ADFS v3.
         //setItemToCacheForUser(resource, clientId, result, null);
     }
 
 
-    private void setItemToCacheForUser(final String issuer, final String resource, final String clientId, final AccessToken accessToken, final RefreshToken refreshToken, final String userId) {
+    private void setItemToCacheForUser(final String issuer, final String resource, final String clientId, final ADALTokenCacheItem cacheItem, final String userId) {
+
+        setItem(CacheKey.createCacheKeyForRTEntry(issuer, resource, clientId, userId), cacheItem);
 
+        if(cacheItem.getIsMultiResourceRefreshToken()){
+            setItem(CacheKey.createCacheKeyForMRRT(issuer, clientId, userId), cacheItem);
+        }
 
+        if(!StringExtensions.isNullOrBlank(cacheItem.getFamilyClientId())){
+            setItem(CacheKey.createCacheKeyForFRT(issuer, clientId, userId), cacheItem);
+        }
 
-/*
-        // new tokens will only be saved into preferred cache location
-        mTokenCacheStore.setItem(CacheKey.createCacheKeyForRTEntry(issuer, resource, clientId, userId),
-                TokenCacheItem.createRegularTokenCacheItem(issuer, resource, clientId, result));
+    }
 
+    private void setItem(String key, ADALTokenCacheItem cacheItem){
 
-        if (result.getIsMultiResourceRefreshToken()) {
-            Log.v(TAG, "Save Multi Resource Refresh token to cache");
-            mTokenCacheStore.setItem(CacheKey.createCacheKeyForMRRT(issuer, clientId, userId),
-                    TokenCacheItem.createMRRTTokenCacheItem(issuer, clientId, result));
+        String json = mGson.toJson(cacheItem);
+        String encrypted = encrypt(json);
+        if (encrypted != null) {
+            mSharedPreferencesFileManager.putString(key, encrypted);
+        } else {
+            Log.e(TAG, "Encrypted output is null");
         }
 
-        // Store separate entries for FRT.
-        if (!StringExtensions.isNullOrBlank(result.getFamilyClientId()) && !StringExtensions.isNullOrBlank(userId)) {
-            Log.v(TAG, "Save Family Refresh token into cache");
-            final TokenCacheItem familyTokenCacheItem = TokenCacheItem.createFRRTTokenCacheItem(issuer, result);
-            mTokenCacheStore.setItem(CacheKey.createCacheKeyForFRT(issuer, result.getFamilyClientId(), userId), familyTokenCacheItem);
+    }
+
+
+    /**
+     * Method that allows to mock StorageHelper class and use custom encryption in UTs.
+     */
+    protected StorageHelper getStorageHelper() {
+        synchronized (LOCK) {
+            if (sHelper == null) {
+                Log.v(TAG, "Started to initialize storage helper");
```

Comment:
nit: assuming we will clean up the logs before we merge to dev

Reply:
We'll have to decide on how we want to do logging for the common library.  I'll add an issue to track this.

Reply:
the reason i called this out is that it will get really noisy if we have logs here

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/13